components/TextFields/src/MDCTextField.h

```diff
@@ -0,0 +1,48 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MDCTextInput.h"
+
+/** Size of an unaltered, Material clear button. */
+extern const CGFloat MDCClearButtonImageSquareSize;
+
+/** When text is manually set via .text or setText:, this notification fires. */
+extern NSString *_Nonnull const MDCTextFieldTextDidSetTextNotification;
+
+/**
+  Material Design themed single line text input.
+  https://www.google.com/design/spec/components/text-fields.html#text-fields-single-line-text-field
+ */
+@interface MDCTextField : UITextField <MDCTextInput>
```

Comment:
We get it from UITextField BUT it has very poor support in ours mostly, I'm guessing, due to the use of IB's TextField instead of IB's View. Does that make any sense?